aws/ec2/helpers.py

```diff
@@ -261,17 +264,25 @@ def ec2_security_group_opens_specific_ports_to_all(
 
 def ec2_instance_test_id(ec2_instance):
     """A getter fn for test ids for EC2 instances"""
-    return "{0[InstanceId]}".format(ec2_instance)
+    return (
+        "{0[InstanceId]}".format(ec2_instance)
+        if isinstance(ec2_instance, dict)
+        else None
+    )
```

Comment:
Couldn't this be:
```python
return get_param_id(ec2_instance, InstanceId)
```
iirc, returning `None` (as opposed to empty string) signals pytest to use its default id generation approach. If we really want to use an empty string as the id, then
```python
return get_param_id(ec2_instance, InstanceId) or ""
```
Should work. (not tested)